File: frameworks/domain.core/src/test/java/com/qomolangma/frameworks/domain/core/CollectionUtilsTest.java
package com.qomolangma.frameworks.domain.core;

import com.google.common.collect.ImmutableList;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.*;

public class CollectionUtilsTest {

    @Test
    void should_normal_input_list() {
        Assertions.assertThat(CollectionUtils.defaultList(ImmutableList.of(1))).isEqualTo(ImmutableList.of(1));
        Assertions.assertThat(CollectionUtils.defaultList(null)).isEqualTo(ImmutableList.of());
    }

    @Test
    void should_return_true() {
        Assertions.assertThat(CollectionUtils.isEmpty(null)).isEqualTo(true);
        Assertions.assertThat(CollectionUtils.isEmpty(new ArrayList<>())).isEqualTo(true);
        Assertions.assertThat(CollectionUtils.notEmpty(List.of("1"))).isEqualTo(true);
        Assertions.assertThat(CollectionUtils.isEmpty(new HashSet<>())).isTrue();
    }

    @Test
    void should_return_false() {
        Assertions.assertThat(CollectionUtils.isEmpty(List.of("1"))).isEqualTo(false);
        Assertions.assertThat(CollectionUtils.notEmpty(null)).isEqualTo(false);
        Assertions.assertThat(CollectionUtils.notEmpty(new ArrayList<>())).isEqualTo(false);
        Assertions.assertThat(CollectionUtils.isNotEmpty(new HashSet<>())).isFalse();
        Assertions.assertThat(CollectionUtils.isNotEmpty(null)).isFalse();
        Assertions.assertThat(CollectionUtils.isEmpty(Set.of("1"))).isFalse();
    }

}

File: frameworks/domain.core/src/test/kotlin/com/qomolangma/frameworks/domain/core/CollectionUtilsTest.kt
package com.qomolangma.frameworks.domain.core

import com.google.common.collect.ImmutableList
import com.qomolangma.frameworks.domain.core.CollectionUtils.defaultList
import com.qomolangma.frameworks.domain.core.CollectionUtils.isEmpty
import com.qomolangma.frameworks.domain.core.CollectionUtils.isNotEmpty
import com.qomolangma.frameworks.domain.core.CollectionUtils.notEmpty
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class CollectionUtilsTest {
    @Test
    fun should_normal_input_list() {
        Assertions.assertThat(defaultList(ImmutableList.of(1))).isEqualTo(ImmutableList.of(1))
        Assertions.assertThat(defaultList<Any>(null)).isEqualTo(ImmutableList.of<Any>())
    }

    @Test
    fun should_return_true() {
        Assertions.assertThat(isEmpty<Any>(null)).isEqualTo(true)
        Assertions.assertThat(isEmpty(ArrayList<Any>())).isEqualTo(true)
        Assertions.assertThat(notEmpty(listOf("1"))).isEqualTo(true)
        Assertions.assertThat(isEmpty(HashSet<Any>())).isTrue
    }

    @Test
    fun should_return_false() {
        Assertions.assertThat(isEmpty(listOf("1"))).isEqualTo(false)
        Assertions.assertThat(notEmpty<Any>(null)).isEqualTo(false)
        Assertions.assertThat(notEmpty(ArrayList<Any>())).isEqualTo(false)
        Assertions.assertThat(isNotEmpty(HashSet<Any>())).isFalse
        Assertions.assertThat(isNotEmpty<Any>(null)).isFalse
        Assertions.assertThat(isEmpty(setOf("1"))).isFalse
    }
}
